File: maxxinke-backend/src/main/java/com/maxxinke/service/AdminService.java
package com.maxxinke.service;

import com.maxxinke.entity.Admin;
import java.util.List;
import java.util.Optional;

/**
 * 管理员服务接口
 * 定义管理员相关的业务操作
 */
public interface AdminService {
    
    /**
     * 创建管理员
     * @param admin 管理员信息
     * @return 创建的管理员
     */
    Admin createAdmin(Admin admin);
    
    /**
     * 更新管理员信息
     * @param id 管理员ID
     * @param admin 管理员信息
     * @return 更新后的管理员
     */
    Admin updateAdmin(Long id, Admin admin);
    
    /**
     * 删除管理员
     * @param id 管理员ID
     */
    void deleteAdmin(Long id);
    
    /**
     * 获取所有管理员列表
     * @return 管理员列表
     */
    List<Admin> findAll();
    
    /**
     * 根据ID获取管理员
     * @param id 管理员ID
     * @return 管理员信息
     */
    Admin getAdminById(Long id);
    
    /**
     * 更新管理员密码
     * @param id 管理员ID
     * @param newPassword 新密码
     */
    void updatePassword(Long id, String newPassword);
    
    /**
     * 根据用户名获取管理员
     * @param username 管理员用户名
     * @return 管理员对象，如果不存在则返回空
     */
    Optional<Admin> getAdminByUsername(String username);
    
    /**
     * 检查用户名是否存在
     * @param username 要检查的用户名
     * @return 存在返回true，否则返回false
     */
    boolean existsByUsername(String username);
    
    /**
     * 验证密码是否正确
     * @param rawPassword 原始密码
     * @param encodedPassword 加密后的密码
     * @return 密码正确返回true，否则返回false
     */
    boolean validatePassword(String rawPassword, String encodedPassword);
    
    /**
     * 对密码进行加密
     * @param rawPassword 原始密码
     * @return 加密后的密码
     */
    String encodePassword(String rawPassword);
    
    /**
     * 获取当前登录的管理员
     * @return 当前管理员对象
     */
    Admin getCurrentAdmin();
} 

File: maxxinke-admin/src/services/adminService.ts
import { adminAPI } from './api';

export interface Admin {
  id: number;
  username: string;
  email?: string;
  createdTime?: string;
  updatedTime?: string;
}

let cachedAdmin: Admin | null = null;

export const adminService = {
  /**
   * 获取当前登录的管理员信息
   * 会缓存结果，减少重复请求
   */
  getCurrentAdmin: async (): Promise<Admin> => {
    if (cachedAdmin) {
      return cachedAdmin;
    }
    
    try {
      const admin = await adminAPI.getCurrentAdmin();
      cachedAdmin = admin;
      return admin;
    } catch (error) {
      console.error('获取当前管理员信息失败:', error);
      throw error;
    }
  },
  
  /**
   * 清除缓存的管理员信息
   * 在登出或修改管理员信息后调用
   */
  clearAdminCache: () => {
    cachedAdmin = null;
  },
  
  /**
   * 更新管理员密码
   * @param adminId 管理员ID
   * @param oldPassword 旧密码
   * @param newPassword 新密码
   */
  updatePassword: async (adminId: number, oldPassword: string, newPassword: string): Promise<void> => {
    try {
      // 先验证旧密码是否正确
      await adminAPI.validatePassword(adminId, oldPassword);
      
      // 再更新新密码
      await adminAPI.updatePassword(adminId, newPassword);
    } catch (error) {
      console.error('修改密码失败:', error);
      throw error;
    }
  }
}; 